web-gateway/src/main/java/controllers/TransactionController.java

```diff
@@ -77,28 +78,29 @@ public static Call transactionsPage(TransactionInfoStatus status, int page, int
-                    CompletionStage<TransactionInfo> transactionFuture = transactionService.getTransaction(itemId).handleRequestHeader(authenticate(user)).invoke();
-                    return transactionFuture.handle((transaction, exception) -> {
-                        if (exception == null) {
-                            Optional<User> seller = Optional.empty();
-                            Optional<User> winner = Optional.empty();
-                            for (User u : nav.getUsers()) {
-                                if (transaction.getCreator().equals(u.getId())) {
-                                    seller = Optional.of(u);
-                                }
-                                if (transaction.getWinner().equals(u.getId())) {
-                                    winner = Optional.of(u);
+                            UUID itemId = UUID.fromString(id);
+                            CompletionStage<TransactionInfo> transactionFuture = transactionService.getTransaction(itemId).handleRequestHeader(authenticate(user)).invoke();
+
+                           return transactionFuture.handle((transaction, exception) -> {
+                               if (exception == null) {
+
+                                   UUID sellerId = transaction.getCreator();
+                                   UUID winnerId = transaction.getWinner();
+                                  getUser(sellerId).thenCombine(getUser(winnerId),
+                                       (seller, winner) -> {
+
+                                           Currency currency = Currency.valueOf(transaction.getItemData().getCurrencyId());
+                                           return ok(views.html.transaction.render(showInlineInstruction, Optional.of(transaction), user, Optional.of(seller), Optional.of(winner), Optional.of(currency), Optional.empty(), (Nav) nav));
+                                       });
+                           } else {
+                                    String msg = exception.getCause().getMessage();
+                                    return ok(views.html.transaction.render(showInlineInstruction, Optional.empty(), user, Optional.empty(), Optional.empty(), Optional.empty(), Optional.of(msg),(Nav) nav));
                                 }
-                            }
-                            Currency currency = Currency.valueOf(transaction.getItemData().getCurrencyId());
-                            return ok(views.html.transaction.render(showInlineInstruction, Optional.of(transaction), user, seller, winner, Optional.of(currency), Optional.empty(), nav));
-                        } else {
-                            String msg = exception.getCause().getMessage();
-                            return ok(views.html.transaction.render(showInlineInstruction, Optional.empty(), user, Optional.empty(), Optional.empty(), Optional.empty(), Optional.of(msg), nav));
+                                return null;
```

Comment:
This is wrong. This makes the compiler happy because the java compiler will fail to check the returned type on all possible branches matches the expectation so it won't complain.